feature_layers/display-route-layer/src/main/java/com/esri/samples/display_route_layer/DisplayRouteLayerSample.java

```diff
@@ -0,0 +1,200 @@
+            portalItem.addDoneLoadingListener(() -> {
+
+                if (portalItem.getLoadStatus() == LoadStatus.LOADED) {
+                    var featureCollection = new FeatureCollection(portalItem);
+
+                    // Create a feature collection layer using the feature collection.
+                    FeatureCollectionLayer featureCollectionLayer = new FeatureCollectionLayer(featureCollection);
+                    // add the feature collection layer to the map's operational layers
+                    mapView.getMap().getOperationalLayers().add(featureCollectionLayer);
+
+                    featureCollectionLayer.addDoneLoadingListener(() -> {
+                        if (featureCollection.getLoadStatus() == LoadStatus.LOADED) {
+                            for (FeatureCollectionTable table : featureCollection.getTables()) {
+                                if (table.getTitle().equals("DirectionPoints")) {
+                                    for (Feature feature : table) {
+                                        String text = (String) feature.getAttributes().get("DisplayText");
+                                        //Creating a Label
+                                        Label label = new Label(text);
+                                        //wrapping the label
+                                        label.setWrapText(true);
+                                        //Setting the alignment to the label
+                                        label.setTextAlignment(TextAlignment.LEFT);
+                                        //Setting the maximum width of the label
+                                        label.setMaxWidth(200);
+                                        //Setting the position of the label
+                                        label.setTranslateX(5);
+                                        label.setTranslateY(5);
+
+
+                                        // create a scroll pane to contain the legend
+                                        //ScrollPane scrollPane = new ScrollPane();
+                                        //scrollPane.setPrefSize(595, 200);
+                                        //scrollPane.setContent(label);
+                                        Platform.runLater(() -> vBox.getChildren().add(label));
+                                        //Platform.runLater(() ->list.setItems(items));
+                                    }
+                                }
+                            }
+                        }
+                    });
+                }
+            });
+
+//          ScrollPane scrollPane = new ScrollPane();
+//          scrollPane.setPrefSize(595, 200);
+//          scrollPane.setContent(label);
+
+
+            // add the map view to the stack pane
+            stackPane.getChildren().addAll(mapView, vBox);
+            StackPane.setAlignment(vBox, Pos.TOP_RIGHT);
+            StackPane.setMargin(vBox, new Insets(10, 10, 0, 0));
```

Comment:
I've created a branch that you can look at to see how you could approach the UI here and get the ScrollPane working. See branch `jen/route-layer-ui` and work through the UI in the code and see how you can get it working.

What I think we need is 3 steps:
1. We create a VBox which we put in the top right of the UI
2. We create a ScrollPane into that VBox
3. We add all the labels into another VBox and put that into the Scrollpane - this means the whole box scrolls down through the labels, rather than adding a ScrollPane to each individual label.

Reply:
<img width="321" alt="scrollpane" src="https://github.com/Esri/arcgis-maps-sdk-java-samples/assets/56556822/09576bb1-d711-40aa-a6de-64e0b513bf30">

Reply:
I referred to this branch` jen/route-layer-ui` and tried creating scrollPane. It is working fine now, thanks for the detailed explanation.